src/webots/app/WbApplication.cpp

```diff
@@ -209,6 +195,10 @@ void WbApplication::loadWorld(QString worldName, bool reloading, bool isLoadingA
 
   worldName = QDir::cleanPath(worldName);
 
+  WbTokenizer tokenizer;
+  tokenizer.tokenize(worldName);
+  WbParser parser(&tokenizer);
```

Comment:
What is the purpose of calling the tokenizer and parser here?

Note that `tokenizer.tokenize(worldName);` is called again at line 224.
And `parser` is used only later at line 241.

Reply:
In the previous backwards compatibility mechanism the tokenizer was used to retrieve the list of undeclared nodes which required moving it to the beginning of the function (since we enter it twice). As this is no longer the case, it can be returned where it was if it's causing issues

Reply:
Yes, this is wrong. I will revert it.